src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,137 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo?: Todo;
+  tempTodo?: Todo;
```

Comment:
```suggestion
  todo: Todo;
```
You can pass the tempTodo as just todo
And todo should be required, without it you don't need to render component at all